packages/@react-stately/data/src/useAsyncList.ts

```diff
@@ -256,21 +307,27 @@ export function useAsyncList<T, C = string>(options: AsyncListOptions<T, C>): As
   // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  let filteredItems = useMemo(
+    () =>  filter ? data.items.filter(item => filter(item, data.filterText)) : data.items,
+    [data.items, data.filterText, filter]);
+
   return {
-    items: data.items,
+    items: filteredItems,
     selectedKeys: data.selectedKeys,
     sortDescriptor: data.sortDescriptor,
-    isLoading: data.state === 'loading' || data.state === 'loadingMore' || data.state === 'sorting',
+    // TODO: add isFiltering? This is so we can have the loading icon in the textfield when filter text changes instead of in the listbox
+    isLoading: data.state === 'loading' || data.state === 'loadingMore' || data.state === 'sorting' || data.state === 'filtering',
```

Comment:
Open question. Reason behind the ask is that the combobox needs to show the progress circle in the input field when loading (whether from filtering causing a new fetch request or just if the menu isn't open) but it also needs to show the progress circle at the bottom of the menu if a `loadMore` is triggered via scrolling. Right now I have the progress circle in the input field always show if `isLoading` is true but this leads to scenarios where you can see both the input field progress circle and the listbox progress circle (e.g. in the client side filtering story if there aren't many items in the list but it is still fetching more).

Reply:
Maybe we should also return the state string and rather than an `isLoading` prop on the ComboBox have a `loadingState` prop? e.g. `loadingState={list.state}` This way the ComboBox could distinguish between the different types of loads.

Reply:
Lemme give that a shot